events/listener.go

```diff
@@ -156,6 +158,93 @@ func (router *EventRouter) run(ready chan<- bool, eventSuffix string) (err error
 	}
 }
 
+func (router *EventRouter) Run() {
```

Comment:
Can you change the name to something more meaningful to the use case it supports and/or add a doc string?

It would be "wrong" to use this style of event consumption for lifecycle process events, so I'd like to warn off developers from using it in ways that they shouldn't.